src/extract/geoboundaries.py

```diff
@@ -34,5 +43,45 @@ def get_geoboundaries_adm(admLevel=None):
         download_url(dl, outpath) 
 
 
-if __name__ == '__main__':
-    get_geoboundaries_adm()
+##
+# Compare with previous json - mainly to keep an eye on the updates
+# End up passing the parameters via the SnakeMake config.
+def get_all_adm0(previous=None):
+    config = parse_yaml('config.yml')
+    url = "{0}?{1}".format(config['geoboundaries']['url'],
+                           config['geoboundaries']['all'])
+    rawdir = os.path.join(config['dirs']['raw_data'],
+                          config['geoboundaries']['subfolder'])
+    geodb_api_response = get_json(url)
+    df = pd.DataFrame(geodb_api_response)
+    if previous:
+        # Load the previous scan/download 
+        # Check the date - if previous is available
+        pass  
+    else:
+        # Use latest CSV with format 'gbd_YYYYMMDD.csv' 
+        # Process the DataFrame to remove all dates that are up to date.
+        # If no latest CSV found download the whole lot
+        for index, row in df.iterrows():
+            # Download the zip (to a GeoBoundaries ADM0 folder)
+            # Unzip is part of the transform process. 
+            # Use transform/adm0.py ? 
+            dl = row['downloadURL']
+            extra, fName = os.path.split(row['downloadURL'])
+            outpath = os.path.join(rawdir, fName)
+            print(r'Downloading {0} to {1}'.format(dl, outpath))
+            download_url(dl, outpath)
```

Comment:
This line fails unless you manually create the directory `raw_data/geoboundaries`. You can add a line that does something like: 
```python
from pathlib import Path
Path(output_dir).mkdir(parents=True, exist_ok=True)
```

Reply:
Ah thanks for the tip! +1